core/src/main/java/me/leoko/advancedban/manager/PunishmentManager.java

```diff
@@ -126,12 +127,33 @@ public void discard(String name) {
      * @return the punishments
      */
     public List<Punishment> getPunishments(String target, PunishmentType put, boolean current) {
+        return put == null ? getPunishments(target, current, PunishmentType.values()) : getPunishments(target, current, put);
+    }
+
+    /**
+     * Get all punishments which belong to the given uuid or ip.
+     *
+     * @param target  the uuid or ip to search for
+     * @param current if only active punishments should be included.
+     * @param put     an array of the basic punishment type to search for ({@link PunishmentType#BAN} would also include Tempbans).
+     *      *                Use <code>null</code> to search for all punishments.
+     * @return the punishments
+     */
+    public List<Punishment> getPunishments(String target, boolean current, PunishmentType ... put ) {
+        MethodInterface mi = Universal.get().getMethods();
         List<Punishment> ptList = new ArrayList<>();
+        List<PunishmentType> putList;
+
+        if(put==null) {
+            putList = new ArrayList<>();
+            mi.getStringList(mi.getConfig(),"FullHistory").forEach((typeString -> putList.add(PunishmentType.valueOf(typeString))));
```

Comment:
I don't think that `put` can be null here, it is just an empty array if no args are provided or `[null]` (an array with null in it) if you pass null. Also this logic should still not be in the API method but rather in the command handler.

Reply:
it can be if you pass a null from the command, or at least it accepts and works that way. However that is a moot point anyways as I am moving the logic to the command as requested